### PR TITLE
Add an icon to the notification API

### PR DESCRIPTION
**Pull request by nickvergessen**
Signed-off-by: Joas Schilling coding@schilljs.com

### CODE REVIEW

**Comment by MorrisJobke** [comment]
:+1:

**Review by MorrisJobke — APPROVED** [review]

**Comment by MorrisJobke** [comment]
cc @rullzer @LukasReschke

File: lib/private/Notification/Notification.php [hunk 1]
```diff
@@ -342,6 +342,28 @@ public function getLink() {
 	}
 
 	/**
+	 * @param string $icon
+	 * @return $this
+	 * @throws \InvalidArgumentException if the icon is invalid
+	 * @since 9.2.0
+	 */
+	public function setIcon($icon) {
+		if (!is_string($icon) || $icon === '' || isset($icon[4000])) {
```
Comment (ChristophWurst) [inline lib/private/Notification/Notification.php]: what does ``isset($icon[4000])`` check?

Comment (MorrisJobke) [inline lib/private/Notification/Notification.php]: That the string is not longer than 400 characters ;)

**Comment by ChristophWurst** [comment]
👍

File: lib/private/Notification/Notification.php (hunk 1 above)
Comment (nickvergessen) [inline lib/private/Notification/Notification.php]: actually not longer than 4000 but i guess you figured that out yourself ;)

**Pull request merged**